utils/youtube.js

```diff
@@ -0,0 +1,51 @@
+const stringSimilarity = require('string-similarity');
+const cheerio = require('cheerio');
+
+const { asyncRequest } = require('./utils.js');
+
+const baseUrl = 'https://www.youtube.com/results?search_query=';
+
+const MATCH_SENSITIVITY = 0.2;
+const INVALID_TITLE_WORDS = /\blyrics\b|\bmirrored\b|\bteaser\b|\btrailer\b|\bdance practice\b/;
+const MV_SCOPING_WORDS = /\bmv\b|\bm\/v\b|\bmusic video\b/;
+
+function _cleanSearchString(str) {
+    return str.toLowerCase().replace(' by ', ' ').replace(' - ', '').replace(/\s\s+/g, ' ').trim();
+};
+
+function _enhanceSearchString(str) {
+    if (MV_SCOPING_WORDS.test(str)) return str;
+    return `${str} mv`;
+};
+
+function _buildYoutubeSearchUrl(str) {
+    const encodedString = encodeURI(str).replace(/\&/g, '%26').replace(/\%20/g, '+');
+    return `${baseUrl}${encodedString}`;
+};
+
+async function getYoutubeUrl(searchString) {
+    const cleanedSearchString = _cleanSearchString(searchString);
+    const enhancedSearchString = _enhanceSearchString(cleanedSearchString);
+    const youtubeSearchUrl = _buildYoutubeSearchUrl(enhancedSearchString);
+    const youtubeResultsPage = await asyncRequest(youtubeSearchUrl);
+    const $ = cheerio.load(youtubeResultsPage);
+    let match;
+
+    $('h3.yt-lockup-title > a').each((index, el) => {
+        const title = $(el).text();
+        const url = `https://www.youtube.com${$(el).attr('href')}`;
+
+        if ((stringSimilarity.compareTwoStrings(title, searchString)) >= MATCH_SENSITIVITY && !(INVALID_TITLE_WORDS.test(title.toLowerCase()))) {
```

Comment:
is it by design to compare this with `searchString` and not `enhancedSearchString`?

Reply:
I think we should probably at least use `cleanedSearchString` in case there's a lot of junk spaces

Reply:
is this why we're not comparing to `enhancedSearchString`? Any reason we shouldn't compared to `cleanedSearchString` though?

Reply:
Oh! I totally thought I changed that to `enhancedSearchString`. We should definitely compare it to that.